graveler/pgiterator.go

```diff
@@ -251,6 +243,11 @@ type PGCommitIterator struct {
 	ctx          context.Context
 	repositoryID RepositoryID
 
+	// initPhase turns true when the iterator was created or `SeekGE()` was called.
+	// initPhase turns false when Next() is called.
+	// When initPhase is true, Value() should return nil.
+	initPhase bool
```

Comment:
why not use the value? it's nil when you initialize, it makes sense to set it to nil when you `SeekGE`